File: groupie-core/src/main/java/com/xwray/groupie/core/GroupDataObserver.java
package com.xwray.groupie.core;

public interface GroupDataObserver {
    void onChanged(Group group);

    void onItemInserted(Group group, int position);

    void onItemChanged(Group group, int position);

    void onItemChanged(Group group, int position, Object payload);

    void onItemRemoved(Group group, int position);

    void onItemRangeChanged(Group group, int positionStart, int itemCount);

    void onItemRangeInserted(Group group, int positionStart, int itemCount);

    void onItemRangeRemoved(Group group, int positionStart, int itemCount);

    void onItemMoved(Group group, int fromPosition, int toPosition);
}

File: library-kotlin/src/main/java/com/wray/groupiekotlin/GroupDataObserver.kt
package com.wray.groupiekotlin

interface GroupDataObserver {
    fun onChanged(group: Group)

    fun onItemInserted(group: Group, position: Int)

    fun onItemChanged(group: Group, position: Int)

    fun onItemChanged(group: Group, position: Int, payload: Any)

    fun onItemRemoved(group: Group, position: Int)

    fun onItemRangeChanged(group: Group, positionStart: Int, itemCount: Int)

    fun onItemRangeChanged(group: Group, positionStart: Int, itemCount: Int, payload: Any)

    fun onItemRangeInserted(group: Group, positionStart: Int, itemCount: Int)

    fun onItemRangeRemoved(group: Group, positionStart: Int, itemCount: Int)

    fun onItemMoved(group: Group, fromPosition: Int, toPosition: Int)
}
